modules/cassandra/src/main/java/org/testcontainers/containers/CassandraContainer.java

```diff
@@ -0,0 +1,158 @@
+                throw new ScriptUtils.UncategorizedScriptException("Error while executing init script: " + initScriptPath, e);
+            }
+        }
+    }
+
+    /**
+     * Map (effectively replace) directory in Docker with the content of resourceLocation if resource location is not null
+     *
+     * Protected to allow for changing implementation by extending the class
+     *
+     * @param pathNameInContainer path in docker
+     * @param resourceLocation    relative classpath to resource
+     */
+    protected void optionallyMapResourceParameterAsVolume(String pathNameInContainer, String resourceLocation) {
+        Optional.ofNullable(resourceLocation)
+                .map(MountableFile::forClasspathResource)
+                .ifPresent(mountableFile -> addFileSystemBind(mountableFile.getResolvedPath(), pathNameInContainer, BindMode.READ_WRITE));
+    }
+
+    /**
+     * Initialize Cassandra with the custom overridden Cassandra configuration
+     * <p>
+     * Be aware, that Docker effectively replaces all /etc/cassandra content with the content of config location, so if
+     * Cassandra.yaml in configLocation is absent or corrupted, then Cassandra just won't launch
+     *
+     * @param configLocation relative classpath with the directory that contains cassandra.yaml and other configuration files
+     */
+    public SELF withConfigurationOverride(String configLocation) {
+        this.configLocation = configLocation;
+        return self();
+    }
+
+    /**
+     * Initialize Cassandra with init CQL script
+     * <p>
+     * CQL script will be applied after container is started (see using WaitStrategy)
+     *
+     * @param initScriptPath relative classpath resource
+     */
+    public SELF withInitScript(String initScriptPath) {
+        this.initScriptPath = initScriptPath;
+        return self();
+    }
+
+    /**
+     * Get username
+     *
+     * By default Cassandra has authenticator: AllowAllAuthenticator in cassandra.yaml
+     * If username and password need to be used, then authenticator should be set as PasswordAuthenticator
+     * (through custom Cassandra configuration) and through CQL with default cassandra-cassandra credentials
+     * user management should be modified
+     */
+    public String getUsername() {
+        return USERNAME;
+    }
+
+    /**
+     * Get password
+     *
+     * By default Cassandra has authenticator: AllowAllAuthenticator in cassandra.yaml
+     * If username and password need to be used, then authenticator should be set as PasswordAuthenticator
+     * (through custom Cassandra configuration) and through CQL with default cassandra-cassandra credentials
+     * user management should be modified
+     */
+    public String getPassword() {
+        return PASSWORD;
+    }
+
+    /**
+     * Get configured Cluster
+     *
+     * Can be used to obtain connections to Cassandra in the container
+     */
+    public Cluster getCluster() {
+        return Cluster.builder()
```

Comment:
I wonder if returning Cluster.Builder is better or not, at least I expect some users to provide additional cluster configuration. WDYT?

Reply:
Cluster.builder() is just static method (obliviously :) ). User can build Cluster instance himself with needed contact points, ports and other configuration.

getCluster() is just convenient method to get configured cluster instance. It can be useful when you have Cassandra only in tests (a bit synthetics case but still) and you don't have configure DataStax driver in you main configuration. Which you would do through system variables and methods getContainerIpAddress() and getMappedPort(...).

So there are two convenient ways to connect Cassandra in the container. No need to bring third with static builder.